### PR TITLE
--version option for the poplog command tool

### PR DESCRIPTION
**Pull request by sfkleach**
This new option reports the GetPoplog version and the version of the base Poplog system. 

Importantly, it defines the single-source of the version, which is a top-level file called `VERSION`. This does not include the leading 'v', which I regard as a human-friendly addition that prevents the version number being [semvar](https://semver.org) compatible.

It incidentally introduces an internal build target `_build/POPLOG_VERSION` that runs corepop and forces a human friendly version of the `pop_internal_version` variable.

Finally, it adds a devteam-only target `github-release`. I have not run this target as yet as, if it works, it will perform a release. Please review.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -611,3 +614,13 @@ buildsnapcraftready:
...
 
+
+################################################################################
+# Perform a GitHub release via CircleCI (you need to be authorised to use this)
+################################################################################
+
+.PHONY: github-release
+github-release: 
+	GETPOPLOG_VERSION=v`cat VERSION`; \
+	git tag $$GETPOPLOG_VERSION -a -m "GetPoplog $$GETPOPLOG_VERSION"
```
Comment (willprice) [inline Makefile]: Doesn't this need a `; \` on the end of the line?

**Review by willprice — COMMENTED** [review]
> Importantly, it defines the single-source of the version, which is a top-level file called VERSION. This does not include the leading 'v', which I regard as a human-friendly addition that prevents the version number being semvar compatible.

Shall we continue tagging with the `v` prefix and keep the version numbers without this? Or would you like to move to having the bare version number for tags moving forward? I have no strong preference either way, but we should document whatever the decision is as part of the release process.

File: makePoplogCommander.sh [hunk 2]
```diff
@@ -238,6 +238,10 @@ UTILITY ACTIONS
 poplog --help
     A special case that shows this usage information.
 
+poplog --version
+    A special case that displays the version information for GetPoplog and
+    the base Poplog system that it accompanies.
```
Comment (willprice) [inline makePoplogCommander.sh]: How about
> Show version information for GetPoplog and the base Poplog system on standard output and exit successfully

Instead?
It's less ambiguous where the information will go and what the exit strategy is.

File: Makefile [hunk 3]
```diff
@@ -611,3 +614,13 @@ buildsnapcraftready:
 	cd _build/dotsnap$(PREBUILT_DIR)/usr/bin; ln -s ../..$(POPLOG_VERSION_DIR)/pop/pop/poplog .
 	cp snapcraft.yaml _build/dotsnap	
 
+
+################################################################################
+# Perform a GitHub release via CircleCI (you need to be authorised to use this)
```
Comment (willprice) [inline Makefile]: > Perform a GitHub release via CircleCI (you need to be authorised to use this

->

> Perform a GitHub release by tagging a commit which triggers CircleCI to build and push binaries for the tag's GitHub release

I'm not sure what the authorized bit is meant to mean? Maybe say "you must be authorized to push tags to the upstream repository on GitHub to perform this action"

File: makePoplogCommander.sh [hunk 4]
```diff
@@ -589,6 +593,14 @@ cat << \****
         } else if ( strcmp( "--help", argv[1] ) == 0 ) {
             printUsage( argc - 2, &argv[2] );
             return EXIT_SUCCESS;
+        } else if ( strcmp( "--version", argv[1] ) == 0 ) {
+****
+
+echo '            printf( "Poplog command tool v'${GET_POPLOG_VERSION:-Undefined}'\\n" );'
+echo '            printf( "Compiled against base Poplog '${POPLOG_VERSION:-Undefined}'\\n" );'
```
Comment (willprice) [inline makePoplogCommander.sh]: Mmmm, I'm not sure about this one. What if I copy `poplog` into a new poplog tree with a different version, then the version string this prints will be wrong. Why don't we invoke `pop11` directly here to get the result? I don't see a good reason not to.

File: Makefile (hunk 1 above)
Comment (sfkleach) [reply Makefile]: Whooops! Fixed

File: makePoplogCommander.sh (hunk 2 above)
Comment (sfkleach) [reply makePoplogCommander.sh]: Done.

File: Makefile (hunk 3 above)
Comment (sfkleach) [reply Makefile]: Done.

File: makePoplogCommander.sh (hunk 4 above)
Comment (sfkleach) [reply makePoplogCommander.sh]: Agreed. Fixed.

**Pull request merged**